packages/edit-navigation/src/components/menu-locations-editor/index.js

```diff
@@ -1,3 +1,117 @@
+/**
+ * External dependencies
+ */
+import { forEach, find, pick } from 'lodash';
+/**
+ * WordPress dependencies
+ */
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState } from '@wordpress/element';
+import {
+	Button,
+	Panel,
+	PanelBody,
+	SelectControl,
+	Spinner,
+} from '@wordpress/components';
+import { __ } from '@wordpress/i18n';
+
+const MenuSelectControl = ( { onChange, menus, menusList, location } ) => {
+	const [ menuId, setMenuId ] = useState( false );
+
+	return (
+		<>
+			{ menus && (
+				<SelectControl
+					options={ menusList }
+					value={ menuId || location.menu }
+					onChange={ ( selectedMenuId ) => {
+						setMenuId( selectedMenuId );
+						onChange( {
+							location: location.name,
+							menu: selectedMenuId,
+						} );
+					} }
+				/>
+			) }
+		</>
+	);
+};
+
 export default function MenuLocationsEditor() {
-	return <>Menu locations editor</>;
+	const menus = useSelect( ( select ) => select( 'core' ).getMenus() );
+	const menuLocations = useSelect( ( select ) =>
+		select( 'core' ).getMenuLocations()
+	);
+	const [ locationsData, setLocationsData ] = useState( {} );
+	const { saveMenu } = useDispatch( 'core' );
+
+	if ( ! menus || ! menuLocations ) {
+		return <Spinner />;
+	}
+
+	const setLocations = ( { location, menu } ) => {
+		locationsData[ location ] = menu;
+		setLocationsData( locationsData );
+	};
+
+	const menusList =
+		menus &&
+		menus.map( ( menu ) => ( {
+			value: menu.id,
+			label: menu.name,
+		} ) );
+
+	menusList.push( {
+		value: 0,
+		label: __( '— Select a Menu —' ),
+	} );
+
+	return (
+		<Panel className="edit-navigation-menu-editor__panel">
+			<PanelBody title={ __( 'Menu locations' ) }>
+				<form
+					onSubmit={ ( e ) => {
```

Comment:
It's probably cleaner if we pull this (very indented) anonymous function out into a named function.